src/Config.h

```diff
@@ -0,0 +1,1488 @@
+
+/**
+ * @file Config.h
+ *
+ * @brief Centralized configuration management with JSON storage
+ */
+
+#pragma once
+
+#include "Logger.h"
+#include "defaults.h"
+#include "hardware/Relay.h"
+#include "hardware/Switch.h"
+#include <ArduinoJson.h>
+#include <LittleFS.h>
+
+class Config {
```

Comment:
On discord you discussed namespaces - which I personally think would be great to introduce - but I don't know what the status is wrt. to that since I have just started reading the PR changes. My suggestion would be to do that in a separate PR since adding a namespace (and its indentation) will generate a full-file diff.